cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,544 @@
+    x and y rotations. See Barends et al., Nature 508, 500 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        q_0: The first qubit under test.
+        q_1: The second qubit under test.
+        num_cfds_seq: The different numbers of Cliffords in the RB study.
+        num_circuits: The number of random circuits generated for each number of
+            Cliffords.
+        num_shots: The number of repetitions of each circuit.
+
+    Returns:
+        A RandomizedBenchmarkingResult object that stores and plots the result.
+    """
+    c1, _, s1, s1_x, s1_y = _single_qubit_cliffords()
+    cfd_matrices = _two_qubit_clifford_matrices(q_0, q_1, c1, s1, s1_x, s1_y)
+    gnd_probs = []
+    for num_cfds in num_cfds_seq:
+        gnd_probs_l = []
+        for _ in range(num_circuits):
+            circuit = _random_two_q_clifford(q_0, q_1, num_cfds, cfd_matrices,
+                                             c1, s1, s1_x, s1_y)
+            circuit.append(ops.measure(q_0, q_1, key='z'))
+            results = sampler.run(circuit, repetitions=num_shots)
+            gnds = [(not r[0] and not r[1]) for r in results.measurements['z']]
+            gnd_probs_l.append(numpy.mean(gnds))
+        gnd_probs.append(float(numpy.mean(gnd_probs_l)))
+
+    return RBResult(num_cfds_seq, gnd_probs)
+
+
+def single_qubit_state_tomography(sampler: sim.SimulatesSamples,
+                                  qubit: devices.GridQubit,
+                                  circuit: circuits.Circuit,
+                                  num_shots: int) -> TomographyResult:
+    """
+    Single-qubit state tomography.
+
+    The density matrix of the output state of a circuit is measured by first
+    doing projective measurements in the z-basis, which determine the
+    diagonal elements of the matrix. A X/2 or Y/2 rotation is then added before
+    the z-basis measurement, which determines the imaginary and real parts of
+    the off-diagonal matrix elements, respectively.
+
+    See Vandersypen and Chuang, Rev. Mod. Phys. 76, 1037 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        circuit: The circuit to execute on the qubit before tomography.
+        num_shots: The number of measurements for each basis rotation.
+
+    Returns:
+        A StateTomographyResult object that stores and plots the density matrix.
+    """
+    circuit_z = circuit + circuits.Circuit.from_ops(ops.measure(qubit, key='z'))
+    results = sampler.run(circuit_z, repetitions=num_shots)
+    rho_11 = numpy.mean(results.measurements['z'])
+    rho_00 = 1.0 - rho_11
+
+    circuit_x = circuit.copy()
+    circuit_x.append(ops.X(qubit) ** 0.5)
+    circuit_x.append(ops.measure(qubit, key='z'))
+    results = sampler.run(circuit_x, repetitions=num_shots)
+    rho_01_im = numpy.mean(results.measurements['z']) - 0.5
+
+    circuit_y = circuit.copy()
```

Comment:
I think this might work:

```python
circuit_y = circuits.Circuit.from_ops(
    circuit,
    ops.Y(qubit)**-0.5,
    ops.measure(qubit, key='z'))
```

Reply:
Cool, it worked. Thanks.